cmd/experimental/kjobctl/pkg/builder/slurm_builder.go

```diff
@@ -524,6 +533,11 @@ func (b *slurmBuilder) buildInitEntrypointScript() (string, error) {
 		memPerGPU = b.memPerGPU.String()
 	}
 
+	nodeList := make([]string, nodes)
+	for i := int32(0); i < nodes; i++ {
+		nodeList[i] = fmt.Sprintf("%s-%d.%s.%s.svc.cluster.local", b.objectName, i, b.objectName, b.namespace)
```

Comment:
Do we need the suffix ".svc.cluster.local"? In this k8s example it is without it https://kubernetes.io/docs/tasks/job/job-with-pod-to-pod-communication/.
I'm concerned this may not work for non-obvious reasons from a user perspective if an admin or cloud-provider has re-configured the cluster. Maybe we should at least make it configurable.

cc @mwielgus

Reply:
That the problem that we discussed. Without this suffix nslookup doesn't allow to get IP correctly.

Reply:
Or maybe we should add this suffix only to get the IP by nslookup.

Reply:
I remember the discussions, and we found that ping doesn't have the issue, so I would suggest using it, or check if nslookup has some extra options.

Hardcoding [cluster-domain](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/) doesn't seem too good. I didn't try changing this myself, but it seems you can specify this for new clusters with terraform for example: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#cluster_dns_domain.

Reply:
I have synced with @mbobrovskyi, another related issue of the initial implementation is that the script runs in the user's main container, rather than init container, so we could guarantee the necessary tools are available on the image. The reason is that the headless service exposes only Pods in the Running phase, whilst a Pod running init container remains in the Pending phase.

One idea we discussed to overcome both challenges is to use kubectl to get the 0-th indexed Pod inside the init container to read the Pod's IP from status field, which is already assigned for Pending Pods. We get get the Pod by listing Pods with selectors for namespace, batch.kubernetes.io/job-completion-index=0 and batch.kubernetes.io/job-name labels (to make it fast). 

One potential difficulty with this approach is to provide auth & authorization for kubectl, hopefully it is not hard, but will probably require extra investigation.